designer/src/state/uiSpec-reducer.ts

```diff
@@ -245,6 +245,50 @@ export const uiSpecificationReducer = createSlice({
         );
       }
     },
+    fieldDuplicated: (
+      state,
+      action: PayloadAction<{
+        originalFieldName: string;
+        newFieldName: string;
+        viewId: string;
+      }>
+    ) => {
+      const {originalFieldName, newFieldName, viewId} = action.payload;
+
+      // check if original field exists
+      if (!(originalFieldName in state.fields)) {
+        throw new Error(
+          `Cannot duplicate unknown field ${originalFieldName} via fieldDuplicated action`
+        );
+      }
+
+      // create a deep copy of the original field
+      const originalField = state.fields[originalFieldName];
+      const newField: FieldType = JSON.parse(JSON.stringify(originalField));
```

Comment:
I guess this is safe since the spec is JSON serialisable - but be careful!! This is a nice hidden bug if you have non serialisable content (speaking from personal experience)